ERCS/erc-7715.md

```diff
@@ -0,0 +1,621 @@
+    W ->>+ U: Show Permission UI
+    U ->>+ W: Approve and sign the permissions object <br> to grant permissions
+    W ->>+ D: Return wallet_grantPermissions response
+    D ->>+ D: Store permissionsContext
+    Note left of D: DApp has permissions context <br> and is looking to send UserOp
+    D ->>+ D: Build UserOp and Sign with Session Key
+    alt Leverage ERC-7679 UserOpBuilder
+    D ->>+ SA: getNonce (with permissionsContext)
+    D ->>+ D: update UserOp with nonce
+    D ->>+ SA: getCallData (with permissionsContext)
+    D ->>+ D: update UserOp with callData
+    D ->>+ SA: getSignature (with permissionsContext)
+    D ->>+ D: update UserOp with signature
+    end
+    D ->>+ D: Send UserOperation
+```
+
+#### ERC-7710
+
+When requesting permissions with a `type` of `account`, the returned data will be redeemable using the interfaces specified in ERC-7710. This allows the recipient of the permissions to use any account type (EOA or contract) to form a transaction or UserOp using whatever payment or relay infrastructure they prefer, by sending an internal message to the returned `permissions.signerMeta.delegationManager` and calling its `function redeemDelegation(bytes calldata _data, Action calldata _action) external;` function with the `_data` parameter set to the returned `permissions.permissionsContext`, and the `_action` data forming the message that the permissions recipient desires the user's account to emit, as defined by this struct:
+
+```
+struct Action {
+    address to;
+    uint256 value;
+    bytes data;
+}
+```
+
+A simple pseudocode example of using a permission in this way, given two ethers signers in the same context, where `alice` wants to request a permission from `bob` might be like this:
+
+```
+// Alice requests a permission from Bob
+const permissionsResponse = await bob.request({
+  method: 'wallet_grantPermissions',
+  params: [{
+    account: bob.address,
+    chainId: 123,
+    signer: {
+      type: 'account',
+      data: {
+        id: alice.address
+      }
+    },
+    permissions: [{
+      type: 'native-token-transfer',
+      policies: [{
+        type: 'gas-limit',
+        data: '0x0186A0'
+      }]
+      data: {
+        allowance: '0x0DE0B6B3A7640000'
+      },
+      required: true
+    }],
+    expiry: Math.floor(Date.now() / 1000) + 3600 // 1 hour from now
+  }]
+});
+
+// Extract the permissionsContext and delegationManager
+const permissionsContext = permissionsResponse.permissionsContext;
+const delegationManager = permissionsResponse.signerMeta.delegationManager;
+
+// Alice forms the action she wants Bob's account to take
+const action = {
+  to: alice.address,
+  value: '0x06F05B59D3B20000'
+  data: '0x'
+};
+
+// Alice sends the transaction by calling redeemDelegation on Bob's account
+const tx = await bob.sendTransaction({
+  to: delegationManager,
+  data: bob.interface.encodeFunctionData('redeemDelegation', [
+    permissionsContext,
+    action
+  ])
+});
```

Comment:
Is this correct?
1. Why `bob.sendTransaction`? shouldn't it be `alice.sendTransaction`?
2. Why `bob` has interface of `redeemDelegation`? `bob` is Delegator, so it has `ERC7710` interface with `executeDelegatedAction` method. `redeemDelegation` lives in the `ERC7710Manager` interface implemented by `delegationManager`.  
[ERC-7710](https://eip.tools/eip/7710) for reference